Content.Shared/Prototypes/EntitySet/EntityListPrototype.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using Robust.Shared.IoC;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.Manager.Attributes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype.List;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Shared.Prototypes.EntitySet
+{
+    [Prototype("entityList")]
+    public class EntityListPrototype : IPrototype
+    {
+        [ViewVariables]
+        [field: DataField("id", required: true)]
+        public string ID { get; } = default!;
+
+        [ViewVariables]
+        [field: DataField("entities", customTypeSerializer: typeof(PrototypeIdListSerializer<EntityPrototype>))]
+        public ImmutableList<string> EntityIds { get; } = ImmutableList<string>.Empty;
+
+        public IEnumerable<EntityPrototype> Entities
+        {
+            get
+            {
+                var prototypeManager = IoCManager.Resolve<IPrototypeManager>();
+
+                foreach (var entityId in EntityIds)
+                {
+                    yield return prototypeManager.Index<EntityPrototype>(entityId);
+                }
+            }
+        }
```

Comment:
IMO make it a method that can have IPrototypeManager passed to it given in your other PR it will be resolved twice.

Reply:
Done in https://github.com/space-wizards/space-station-14/pull/3720/commits/cdc170be5a7cbb3b79d6fd5f662739d5756a5de8